README.rst

```diff
@@ -0,0 +1,208 @@
+2012.
+
+Analysis structure
+==================
+
+Making a research data analysis reproducible basically means to provide
+"runnable recipes" addressing (1) where is the input data, (2) what software was
+used to analyse the data, (3) which computing environments were used to run the
+software and (4) which computational workflow steps were taken to run the
+analysis. This will permit to instantiate the analysis on the computational
+cloud and run the analysis to obtain (5) output results.
+
+
+1. Input data
+-------------
+
+The analysis takes the following inputs:
+
+- the list of CMS validated runs included in the ``inputs`` directory:
+
+  - ``Cert_190456-208686_8TeV_22Jan2013ReReco_Collisions12_JSON.txt``
+
+- a set of official CMS data files in the `ROOT <https://root.cern.ch/>`_ format
+  included in the ``inputs`` directory:
+
+  - ``DoubleE11.root``
+  - ``DoubleE12.root``
+  - ``DoubleMu11.root``
+  - ``DoubleMu12.root``
+  - ``DY1011.root``
+  - ``DY1012.root``
+  - ``DY101Jets12.root``
+  - ``DY50Mag12.root``
+  - ``DY50TuneZ11.root``
+  - ``DY50TuneZ12.root``
+  - ``DYTo2mu12.root``
+  - ``HZZ11.root``
+  - ``HZZ12.root``
+  - ``TTBar11.root``
+  - ``TTBar12.root``
+  - ``TTJets11.root``
+  - ``TTJets12.root``
+  - ``ZZ2mu2e11.root``
+  - ``ZZ2mu2e12.root``
+  - ``ZZ4e11.root``
+  - ``ZZ4e12.root``
+  - ``ZZ4mu11.root``
+  - ``ZZ4mu12.root``
+
+- CMS collision data from 2011 and 2012 accessed "live" during analysis via
+  `CERN Open Data <http://opendata.cern.ch/>`_ portal:
+
+  - `/DoubleMuParked/Run2012C-22Jan2013-v1/AOD <http://opendata.cern.ch/record/6030>`_
+
+- CMS simulated data from 2011 and 2012 accessed "live" during analysis via
+  `CERN Open Data <http://opendata.cern.ch/>`_ portal:
+
+  - `/SMHiggsToZZTo4L_M-125_8TeV-powheg15-JHUgenV3-pythia6/Summer12_DR53X-PU_S10_START53_V19-v1/AODSIM <http://opendata.cern.ch/record/9356>`_
+
+2. Analysis code
+----------------
+
+The analysis will consist of two stages. In the first stage, we shall process
+the original collision data (using ``demoanalyzer_cfg_level3data.py``) and
+simulated data (using ``demoanalyzer_cfg_level3MC.py``) for one Higgs signal
+candidate with with reduced statistics. In the second stage, we shall plot the
+results (using ``M4Lnormdatall_lvl3.cc``). The ``HiggsDemoAnalyzer`` directory
+contains the analysis code plugin for the `CMSSW <http://cms-sw.github.io/>`_
+analysis framework.
+
+3. Compute environment
+----------------------
+
+In order to be able to rerun the analysis even several years in the future, we
+need to "encapsulate the current compute environment", for example to freeze the
+software package versions our analysis is using. We shall achieve this by
+preparing a `Docker <https://www.docker.com/>`_ container image for our analysis
+steps.
+
+This analysis example runs within the `CMSSW <http://cms-sw.github.io/>`_
+analysis framework that was packaged for Docker in `clelange/cmssw
+<https://hub.docker.com/r/clelange/cmssw/>`_.
```

Comment:
The actual version (`CMSSW_5_3_32`) might be interesting to note (and the image tag, `clelange/cmssw_5_3_32`).